pilot/pkg/xds/ads.go

```diff
@@ -435,27 +441,21 @@ func (s *DiscoveryServer) shouldRespond(con *Connection, request *discovery.Disc
 	previousInfo.AlwaysRespond = false
 	con.proxy.Unlock()
 
-	// Envoy can send two DiscoveryRequests with same version and nonce
+	// We should always respond "alwayRespond" marked requests to let Envoy finish warming
+	// even though Nonce match and it looks like an ACK.
```

Comment:
> and it looks like an ACK.

Is not quite right -- it could be legitimate resource change? Which may give confusing logs?

Not likely to happen much/ever in practice though. But it may be more accurate if we only log this inside `len(removed) == 0 && len(added) == 0`